### ISSUE TITLE
Unit tests

### ISSUE THREAD

**Issue by tsyesika** [xray7224/MegBot#52]
While these are a low priority right now it's easily to underestimate their importance. MegBot could really start to benefit these to be put over the libraries, core plugins and even possibly the user space plugins.

The benefit they bring I see us using them in a commit hook which check's we haven't broken anything before commiting. They also allow us to easily do sainty check so that what we think things do, they actually do.

I believe python already has a unit testing framework (what about a mocking framework?) this should be looked into before starting
